packages/hasura-storage-js/src/hasura-storage-client.ts

```diff
@@ -33,29 +35,58 @@ export class HasuraStorageClient {
   }
 
   /**
-   * Use `nhost.storage.upload` to upload a file. The `file` must be of type [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File).
+   * Use `nhost.storage.upload` to upload a file. 
+   * 
+   * It's possible to use [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File) or [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) to upload a file. The `File` instance is only available in the browser while `FormData` with [`form-data`](https://www.npmjs.com/package/form-data) works both in the browser and in NodeJS (server).
    * 
-   * If no `bucket` is specified the `default` bucket will be used.
+   * If no `bucketId` is specified the bucket `default` is used.
    *
    * @example
+   * 
+   * Example to upload a file from the browser using `File`.
+   * 
    * ```ts
    * await nhost.storage.upload({ file })
    * ```
+   * 
+   * Example to upload a file from the browser using `File` to a specific Bucket.
    * 
     @example
    * ```ts
    * await nhost.storage.upload({ file, bucketId: '<Bucket-ID>' })
    * ```
+   * 
+   * Example to upload a file using `FormData` from the server.
    *
+   * @example
+   * ```ts
+   * import FormData from 'form-data'
+   * 
+   * const fd = new FormData() 
+   * fd.append('file', fs.createReadStream('./tests/assets/sample.pdf'))
+   * 
+   * await storage.upload({
+   *   formData: fd
+   * })
+   * ```
+   * 
    * @docs https://docs.nhost.io/reference/javascript/storage/upload
    */
   async upload(params: StorageUploadParams): Promise<StorageUploadResponse> {
```

Comment:
I think you should create two [overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads) for `upload`:

```ts
/**
 * ... first documentation ...
 */
upload(params: StorageUploadFileParams): Promise<StorageUploadResponse>;

/**
 * ... second documentation ...
 */
upload(params: StorageUploadFormDataParams): Promise<StorageUploadResponse>;

async upload(params: StorageUploadFileParams | StorageUploadFormDataParams): Promise<StorageUploadResponse> {
  // you'll still need to use a type check similar to what you are doing in the function body
}
```

This would enhance the IDE experience (note the 1/2 label in the tooltip) and would also let us create separate documentation for these functions. 
![Screenshot 2022-08-24 at 16 09 51](https://user-images.githubusercontent.com/310881/186440395-87c55a39-72d2-4cb9-a533-9ec7ebb218d8.png)

Actually I think we could further improve this API by changing the function signature to this:
```ts
/**
 * ... first documentation ...
 */
upload(file: File, params: StorageUploadParams): Promise<StorageUploadResponse>;

/**
 * ... second documentation ...
 */
upload(formData: FormData, params: StorageUploadParams): Promise<StorageUploadResponse>;

async upload(fileOrFormData: File | FormData, params: StorageUploadParams) {
  // ...
}
```

This however requires a bigger change from our users on their end, but it would make the parameter list more logical IMO.